src/PrestaShopBundle/Resources/config/services/core/grid/grid_definition_factory.yml

```diff
@@ -198,6 +198,15 @@ services:
             - '@prestashop.core.form.choice_provider.order_state_by_id'
         public: true
 
+    prestashop.core.grid.definition.factory.cart_rule:
+        class: 'PrestaShop\PrestaShop\Core\Grid\Definition\Factory\CartRuleGridDefinitionFactory'
+        parent: 'prestashop.core.grid.definition.factory.abstract_grid_definition'
+        arguments:
+          - '@request_stack'
```

Comment:
I don't think you are using these arguments in factory. Double check if not forgotten from copy/paste.